src/widgets/vitals/vitals-overview.component.tsx

```diff
@@ -84,6 +84,13 @@ const VitalsOverview: React.FC<VitalsOverviewProps> = () => {
     }
   }, [patientUuid]);
 
+  React.useEffect(() => {
+    if (!isEmpty(vitals)) {
+      const [page, allPages] = paginate<any>(vitals, pageNumber, pageSize);
```

Comment:
yea, same here - i think if you implement this with useMemo, i'm not sure you would need state for currentPage / setCurrentPage